aidies-project/client/src/components/CampaignEdit.js

```diff
@@ -0,0 +1,240 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import { editCampaign, getCharities, getCategories } from '../store/campaign';
+import { makeStyles } from '@material-ui/core/styles';
+import { Card, Typography, Button, FormControl, Grid, InputLabel, Select, MenuItem, TextField, IconButton } from '@material-ui/core';
+import PhotoCamera from '@material-ui/icons/PhotoCamera';
+import '../css/campaign.css';
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        '& > *': {
+            margin: theme.spacing(1),
+        },
+    },
+    details: {
+        display: 'flex',
+        flexDirection: 'column',
+    },
+    input: {
+        display: 'none',
+    },
+    textField: {
+        marginLeft: theme.spacing(1),
+        marginRight: theme.spacing(1),
+    },
+    button: {
+        backgroundColor: '#222',
+        color: '#fff',
+        textAlign: 'right',
+    },
+    formControl: {
+        minWidth: 502,
+    },
+    inputLabel: {
+        paddingTop: -5,
+        paddingLeft: 14,
+    }
+}));
+const CampaignCreate = () => {
+    const [campaignName, setCampaignName] = useState('New awesome campaign');
+    const [summary, setSummary] = useState('new awesome summary');
+    const [story, setStory] = useState('great story');
+    const [image, setImage] = useState('');
+    const [startingPrice, setStartingPrice] = useState('50');
+    const [closingDate, setClosingDate] = useState('2020-10-26T18:36');
+    const [charity, setCharity] = useState(2);
+    const [category, setCategory] = useState(3);
+    const [button, setButton] = useState('Start a campaign');
+
+    const classes = useStyles();
+    const history = useHistory();
+    const dispatch = useDispatch();
+
+    const campaign = useSelector(state => state.campaign.current);
+    const userId = useSelector(state => state.authentication.user.id);
+    const charities = useSelector(state => state.campaign.charities);
+    const categories = useSelector(state => state.campaign.categories);
+    
+    useEffect(() => {
+        dispatch(getCategories());
+        dispatch(getCharities());
+        setCampaignName(campaign.name);
+        setSummary(campaign.summary);
+        setStory(campaign.story);
+        setImage(campaign.image);
+        setStartingPrice(campaign.startingPrice);
+        const editedDate = campaign.closingDate.slice(0,campaign.closingDate.length-2);
+        setClosingDate(editedDate);
+        setCharity(campaign.charityId);
+        setCategory(campaign.categoryId);
+        setButton('Submit Changes');
```

Comment:
Hmm, seeing how long this is I think you should refactor this component to just have an `updatedCampaign` state. Then, in order to update the slice of state, you would make use of the `previousState` to spread the other key/value pairs and update a specific key/value pair by overwriting it:

```js
setUpdatedCampaign(prevState => ({
    ...prevState,
    [name]: value
}));
```

You can check out this [Stack Overflow post](https://stackoverflow.com/questions/54150783/react-hooks-usestate-with-object) for more!